### ISSUE TITLE
Can't generate most of the files due to memory overflow

### ISSUE THREAD

**Issue by guest123guest** [commanderx16/x16-rom#336]
kernal/drivers/x16/joystick.s:214: Warning: No reference to unnamed label
kernal/drivers/x16/joystick.s:217: Warning: No reference to unnamed label
kernal/drivers/x16/joystick.s:220: Warning: No reference to unnamed label
ld65: Warning: build/x16/cfg/kernal-x16.cfg:87: Segment 'NVRAM' does not exist
ld65: Warning: build/x16/cfg/kernal-x16.cfg:33: Segment 'KERNRAM2' overflows memory area 'KERNAL' by 2 bytes
ld65: Error: Cannot generate most of the files due to memory area overflow 
make: *** [Makefile:385: build/x16/kernal.bin] Error 1

How do I fix this?

**Comment by irmen** [comment]
unconfirmed but this looks like a buggy cc65 installation. What version is yours? (ld65 --version)  Mine is ld65 V2.18 - Git ece63f0

**Comment by guest123guest** [comment]
ld65 V2.19 - Git 71d64a3c1

**Comment by irmen** [comment]
ok well, I don't know then :-(

**Comment by guest123guest** [comment]
Should I downgrade?

**Comment by irmen** [comment]
I really don't know, the important thing is , from what I know, that you got a build that includes the X16 support in it.  Otherwise I know very little about CC65 toolchain, so let's hope someone more knowledgeable comes along here.

**Comment by dmsc** [comment]
Hi!

With r41, assembly works:

```
x16-rom$ git checkout r41
HEAD is now at a448ed3 [KEYMAP] ABC/X16 Shift+Alt+k is SHY (X16 logo)
x16-rom$ make
make: Nothing to be done for 'all'.
```

With current head, it does not work:
```
x16-rom$ git checkout master
Previous HEAD position was a448ed3 [KEYMAP] ABC/X16 Shift+Alt+k is SHY (X16 logo)
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

x16-rom$ make
ca65 -I inc -D bsw=1 -D drv1541=1 -I geos/inc -I geos -D CPU_65C02=1  -g -D MACHINE_X16=1 --cpu 65SC02 -l build/x16/kernal/drivers/x16/i2c.lst kernal/drivers/x16/i2c.s -o build/x16/kernal/drivers/x16/i2c.o
ca65 -I inc -D bsw=1 -D drv1541=1 -I geos/inc -I geos -D CPU_65C02=1  -g -D MACHINE_X16=1 --cpu 65SC02 -l build/x16/kernal/drivers/x16/smc.lst kernal/drivers/x16/smc.s -o build/x16/kernal/drivers/x16/smc.o
ca65 -I inc -D bsw=1 -D drv1541=1 -I geos/inc -I geos -D CPU_65C02=1  -g -D MACHINE_X16=1 --cpu 65SC02 -l build/x16/kernal/drivers/x16/rtc.lst kernal/drivers/x16/rtc.s -o build/x16/kernal/drivers/x16/rtc.o
ld65 -C build/x16/cfg/kernal-x16.cfg build/x16/kernal/declare.o build/x16/kernal/vectors.o build/x16/kernal/kbdbuf.o build/x16/kernal/memory.o build/x16/kernal/lzsa.o build/x16/kernal/cbm/editor.o build/x16/kernal/cbm/channel/channel.o build/x16/kernal/cbm/init.o build/x16/kernal/cbm/memory.o build/x16/kernal/cbm/nmi.o build/x16/kernal/cbm/irq.o build/x16/kernal/cbm/util.o build/x16/kernal/cbm/serial.o build/x16/kernal/drivers/x16/x16.o build/x16/kernal/drivers/x16/memory.o build/x16/kernal/drivers/x16/screen.o build/x16/kernal/drivers/x16/ps2.o build/x16/kernal/drivers/x16/ps2kbd.o build/x16/kernal/drivers/x16/ps2mouse.o build/x16/kernal/drivers/x16/joystick.o build/x16/kernal/drivers/x16/clock.o build/x16/kernal/drivers/x16/rs232.o build/x16/kernal/drivers/x16/framebuffer.o build/x16/kernal/drivers/x16/sprites.o build/x16/kernal/drivers/x16/entropy.o build/x16/kernal/drivers/x16/beep.o build/x16/kernal/drivers/x16/i2c.o build/x16/kernal/drivers/x16/smc.o build/x16/kernal/drivers/x16/rtc.o build/x16/kernal/drivers/generic/softclock_timer.o build/x16/kernal/graph/graph.o build/x16/kernal/fonts/fonts.o build/x16/kernal/graph/console.o build/x16/kernal/ieee_switch.o -o build/x16/kernal.bin -m build/x16/kernal.map -Ln build/x16/kernal.sym
ld65: Warning: build/x16/cfg/kernal-x16.cfg:87: Segment 'NVRAM' does not exist
ld65: Warning: build/x16/cfg/kernal-x16.cfg:33: Segment 'KERNRAM2' overflows memory area 'KERNAL' by 2 bytes
ld65: Error: Cannot generate most of the files due to memory area overflow 
make: *** [Makefile:385: build/x16/kernal.bin] Error 1
```

With previous-to-last revision, it works:
```
x16-rom$ git checkout 647d4f3efd4dc2828fce366ef5c5d99eac5dd0a3
Note: switching to '647d4f3efd4dc2828fce366ef5c5d99eac5dd0a3'.
HEAD is now at 647d4f3 [README] Made keywords into markdown code (#325)

x16-rom$ make
ca65 -I inc -D bsw=1 -D drv1541=1 -I geos/inc -I geos -D CPU_65C02=1  -g -D MACHINE_X16=1 --cpu 65SC02 -l build/x16/kernal/drivers/x16/rtc.lst kernal/drivers/x16/rtc.s -o build/x16/kernal/drivers/x16/rtc.o
ld65 -C build/x16/cfg/kernal-x16.cfg build/x16/kernal/declare.o build/x16/kernal/vectors.o build/x16/kernal/kbdbuf.o build/x16/kernal/memory.o build/x16/kernal/lzsa.o build/x16/kernal/cbm/editor.o build/x16/kernal/cbm/channel/channel.o build/x16/kernal/cbm/init.o build/x16/kernal/cbm/memory.o build/x16/kernal/cbm/nmi.o build/x16/kernal/cbm/irq.o build/x16/kernal/cbm/util.o build/x16/kernal/cbm/serial.o build/x16/kernal/drivers/x16/x16.o build/x16/kernal/drivers/x16/memory.o build/x16/kernal/drivers/x16/screen.o build/x16/kernal/drivers/x16/ps2.o build/x16/kernal/drivers/x16/ps2kbd.o build/x16/kernal/drivers/x16/ps2mouse.o build/x16/kernal/drivers/x16/joystick.o build/x16/kernal/drivers/x16/clock.o build/x16/kernal/drivers/x16/rs232.o build/x16/kernal/drivers/x16/framebuffer.o build/x16/kernal/drivers/x16/sprites.o build/x16/kernal/drivers/x16/entropy.o build/x16/kernal/drivers/x16/beep.o build/x16/kernal/drivers/x16/i2c.o build/x16/kernal/drivers/x16/smc.o build/x16/kernal/drivers/x16/rtc.o build/x16/kernal/drivers/generic/softclock_timer.o build/x16/kernal/graph/graph.o build/x16/kernal/fonts/fonts.o build/x16/kernal/graph/console.o build/x16/kernal/ieee_switch.o -o build/x16/kernal.bin -m build/x16/kernal.map -Ln build/x16/kernal.sym
ld65: Warning: build/x16/cfg/kernal-x16.cfg:87: Segment 'NVRAM' does not exist
./scripts/relist.py build/x16/kernal.map build/x16/kernal
INFO: Could not find segment matching 'CLOCK' in module clock, assuming zero offset
INFO: Could not find segment matching 'KVAR' in module smc, assuming zero offset
cat build/x16/kernal.bin build/x16/keymap.bin build/x16/dos.bin build/x16/geos.bin build/x16/basic.bin build/x16/monitor.bin build/x16/charset.bin build/x16/codex.bin build/x16/demo.bin > build/x16/rom.bin
./scripts/symbolize.sh 0 build/x16/kernal.sym   > build/x16/rom_labels.h
./scripts/symbolize.sh 1 build/x16/keymap.sym  >> build/x16/rom_labels.h
./scripts/symbolize.sh 2 build/x16/dos.sym     >> build/x16/rom_labels.h
./scripts/symbolize.sh 3 build/x16/geos.sym    >> build/x16/rom_labels.h
./scripts/symbolize.sh 4 build/x16/basic.sym   >> build/x16/rom_labels.h
./scripts/symbolize.sh 5 build/x16/monitor.sym >> build/x16/rom_labels.h
./scripts/symbolize.sh 6 build/x16/charset.sym >> build/x16/rom_labels.h
./scripts/trace_lst.py 0 `find build/x16/kernal/ -name \*.rlst` > build/x16/rom_lst.h
./scripts/trace_lst.py 2 `find build/x16/dos/ -name \*.rlst`   >> build/x16/rom_lst.h
./scripts/trace_lst.py 3 `find build/x16/geos/ -name \*.rlst`   >> build/x16/rom_lst.h
./scripts/trace_lst.py 4 `find build/x16/basic/ -name \*.rlst` >> build/x16/rom_lst.h
./scripts/trace_lst.py 5 `find build/x16/monitor/ -name \*.rlst`   >> build/x16/rom_lst.h
```

So, with commit 8b5ac95a84499259bbc3a82d1058ff4ba87a5442  code does not fit in the ROM area anymore.

Have Fun!

**Comment by guest123guest** [comment]
Checked out the master, it seems to work now? I'm not sure if it should be closed yet though.

**Comment by irmen** [comment]
#334 has been merged there ought to be *heaps* of space available again in the rom bank now

**Comment by dmsc** [comment]
So, this is fixed with dd6623727b1fa8468f6a5568f49c3bbde23114f7

**Comment by guest123guest** [comment]
I guess I can close this issue for now.
